concordium-consensus/src/Concordium/GlobalState/Basic/TreeState.hs

```diff
@@ -226,19 +234,29 @@ instance (bs ~ BlockState m, BS.BlockStateStorage m, Monad m, MonadIO m, MonadSt
 
     getCredential txHash =
       preuse (transactionTable . ttHashMap . ix txHash) >>= \case
-        Just (WithMetadata{wmdData=CredentialDeployment{..},..}, _) -> return $ Just WithMetadata{wmdData=biCred,..}
+        Just (WithMetadata{wmdData=CredentialDeployment{..},..}, status) ->
+          case status of
+            Received _ verRes -> return $ Just (WithMetadata{wmdData=biCred,..}, verRes)
+            Committed _ _ verRes -> return $ Just (WithMetadata{wmdData=biCred,..}, verRes)
+            _ -> return $ Just (WithMetadata{wmdData=biCred,..}, Nothing)
         _ -> return Nothing
     
-    getNonFinalizedChainUpdates uty sn =
+    getNonFinalizedChainUpdates uty sn = do
+      table <- use transactionTable
       use (transactionTable . ttNonFinalizedChainUpdates . at' uty) >>= \case
         Nothing -> return []
         Just nfcus ->
           let (_, atsn, beyond) = Map.splitLookup sn (nfcus ^. nfcuMap)
           in return $ case atsn of
-            Nothing -> Map.toAscList beyond
-            Just s -> (sn, s) : Map.toAscList beyond
+            Nothing ->
+              let withStatuss = Map.map (Set.map (\tx -> withStatus tx table)) beyond
+              in Map.toAscList withStatuss
+            Just s ->
+              let first = (sn, Set.map (\tx -> withStatus tx table) s)
+                  rest = Map.toAscList $ Map.map (Set.map (\tx -> withStatus tx table)) beyond
+              in first : rest
 
-    addCommitTransaction bi@WithMetadata{..} slot = do
+    addCommitTransaction (bi@WithMetadata{..}, verRes) slot = do
```

Comment:
Any reason this is a pair as opposed to two arguments?

Reply:
It's a `BlockItemWithStatus`